esmlab/statistics.py

```diff
@@ -358,4 +359,42 @@ def weighted_corr(x, y, weights=None, dim=None, apply_nan_mask=True):
         * weighted_cov(y, y, weights=weights, dim=op_over_dims, apply_nan_mask=apply_nan_mask_flag)
     )
     corr_xy = numerator / denominator
-    return corr_xy
+
+    if return_p:
```

Comment:
@bradyrx, should we return the p-value as an attribute?

Reply:
I think it's best as a DataArray for two reasons:
1. One can correlate two gridded products, or a time series to a grid, and then be returned an array of p-values corresponding to grid cells. (I'm not aware of any good way to do this via attributes)
2. Using a DataArray allows one to mask with `where` since the coordinates are one-to-one. E.g., `ds.where(p < 0.05)`. 

I'll push forth on this PR later this week with the effective sample sizes. It's also probably worth waiting for the `dim` keyword fix before merging. Let me know your thoughts.

Reply:
Good points. But this requires two computations to get the significance.

Multiple (optional) return arguments?  Return a Dataset?

Reply:
> But this requires two computations to get the significance.

Could you clarify where you're looking to reduce computations? As a base case, `corr_xy` will always be computed. To get significance, we have to compute `t` no matter what (which is a quick vectorized numpy calculation so should have low overhead). Then also seems like we're constrained to `scipy` for the t-test.

> Multiple (optional) return arguments? Return a Dataset?

I was thinking a Dataset might be the most straight forward to avoid the nuisance of returning a separate `r` and `p`.  Although this probably means `weighted_corr` can only be run on DataArrays, since a Dataset would imply separate p-values for each variable which could make naming variables difficult.

In a DataArray only framework, it would look like:
```python
esmlab.statistics.weighted_corr(x, y, dim='time')  # where x, y is gridded lat/lon

<xarray.DataArray (lat: 10, lon: 10)>
r (lat, lon) ...
p (lat, lon) ...
```

Thanks for iterating on this. Simple addition but worth thinking about all the use cases and seems like it might dictate some philosophy changes to the package.